src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -1309,14 +1309,16 @@ private static Attribute NewValidateSetAttribute(AttributeAst ast)
             // 'ValidateSet([CustomGeneratorType], IgnoreCase=$false)' is supported in scripts.
             if (ast.PositionalArguments.Count == 1 && ast.PositionalArguments[0] is TypeExpressionAst generatorTypeAst)
             {
-                if (TypeResolver.TryResolveType(generatorTypeAst.TypeName.FullName, out Type generatorType))
+                var generatorType = TypeResolver.ResolveITypeName(generatorTypeAst.TypeName, out Exception exception);
```

Comment:
`ResolveITypeName` again check `iTypeName as TypeName` - can we use `internal static Type ResolveTypeName(TypeName typeName, out Exception exception)`?

Reply:
No, you might have something else like a generic or array type instead of a simple type.

Reply:
Thanks for clarify!
Should we test these cases too (generic or array)?

Reply:
The ast must be fully general which explains why you must use ResolveITypeName here.

In this specific case, it is extremely unlikely anyone would ever use a generic (which should work) or an array (which should be an error), so testing has marginal usefulness.